File: AICodeDetector/datasets_mask_fill/go/train2/AI/binarySearch2.go
package main

import (
	"fmt"
)

// binarySearchIterative performs binary search on a sorted array and returns the index of the target if found, else -1.
func binarySearchIterative(arr []int, target int) int {
	low := 0
	high := len(arr) - 1

	for low <= high {
		mid := low + (high-low)/2
		if arr[mid] == target {
			return mid
		} else if arr[mid] < target {
			low = mid + 1
		} else {
			high = mid - 1
		}
	}
	return -1
}

func main() {
	arr := []int{2, 5, 8, 12, 16, 23, 38, 56, 72, 91}
	target := 23

	result := binarySearchIterative(arr, target)
	if result != -1 {
		fmt.Printf("Element found at index: %d\n", result)
	} else {
		fmt.Println("Element not found in the array.")
	}
}

File: AICodeDetector/datasets/Python/train/AI/binary_search_2.py
def binary_search_recursive(arr, target, low, high):
    if low > high:
        return None  # 要素が見つからない場合

    mid = (low + high) // 2
    guess = arr[mid]

    if guess == target:
        return mid  # 目的の要素が見つかった場合
    elif guess > target:
        return binary_search_recursive(arr, target, low, mid - 1)  # 下半分を探索
    else:
        return binary_search_recursive(arr, target, mid + 1, high)  # 上半分を探索

# テストコード
if __name__ == '__main__':
    my_list = [1, 3, 5, 7, 9, 11, 13, 15, 17]
    print(binary_search_recursive(my_list, 3, 0, len(my_list) - 1))  # 出力: 1
    print(binary_search_recursive(my_list, -1, 0, len(my_list) - 1)) # 出力: None
